### PR TITLE
EZP-31121: Update the recommendation bundle for v3

### PR DESCRIPTION
**Pull request by ciastektk**
https://jira.ez.no/browse/EZP-31121

### CODE REVIEW

File: src/bundle/Controller/RecommendationController.php [hunk 1]
```diff
@@ -84,29 +74,26 @@ public function showRecommendationsAction(Request $request): Response
...
     {
         $this->sendDeliveryFeedback = $value;
     }
 
-    /**
-     * @param string|null $template
-     *
-     * @return string
-     */
+    protected function getTwig(): Environment
```
Comment (damianz5) [inline src/bundle/Controller/RecommendationController.php]: could be changed to: ([ref](https://symfony.com/doc/current/service_container/service_subscribers_locators.html))
```
protected function twig(): Environment
{
    return $this->container->get(__METHOD__);
}
```

File: src/bundle/Templating/Twig/Functions/AbstractFunction.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzRecommendationClientBundle\Templating\Twig\Functions;
+
+use Twig\Environment as TwigEnvironment;
+
+abstract class AbstractFunction
```
Comment (damianz5) [inline src/bundle/Templating/Twig/Functions/AbstractFunction.php]: Consider changing to `trait TwigAware` and use `ServiceSubscriberTrait & ServiceSubscriberInterface`.

File: src/lib/API/AbstractAPI.php [hunk 3]
```diff
@@ -58,21 +48,16 @@ protected function buildEndPointUri(array $endPointParameters, ?string $rawEndPo
...
         $queryString = '';
 
         foreach ($parameters as $parameterKey => $parameterValue) {
-            if (is_array($parameterValue)) {
+            if (\is_array($parameterValue)) {
                 $queryString .= $this->buildQueryStringFromArray($parameterValue);
             }
 
-            if (is_string($parameterValue) || is_numeric($parameterValue)) {
+            if (\is_string($parameterValue) || is_numeric($parameterValue)) {
```
Comment (damianz5) [inline src/lib/API/AbstractAPI.php]: `\is_numeric`

File: src/lib/Event/Listener/LoginListener.php [hunk 4]
```diff
@@ -43,11 +43,6 @@ class LoginListener
     private $logger;
 
     /**
-     * @param \Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface $authorizationChecker
-     * @param \Symfony\Component\HttpFoundation\Session\SessionInterface $session
-     * @param \EzSystems\EzRecommendationClient\Client\EzRecommendationClientInterface $client
-     * @param \eZ\Publish\API\Repository\UserService $userService
-     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
      * @param \Psr\Log\LoggerInterface $logger
```
Comment (damianz5) [inline src/lib/Event/Listener/LoginListener.php]: all dependencies should be required (remove `?` from `?LoggerInterface`) NullLogger could be used if needed

File: src/lib/Value/ExportParameters.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * @var int
+     *
+     * @Assert\Positive
+     */
+    public $pageSize;
+
+    /**
+     * @var int
+     *ExportRequestMapper
```
Comment (damianz5) [inline src/lib/Value/ExportParameters.php]: is it `ExportRequestMapper`?

File: src/bundle/Templating/Twig/Functions/AbstractFunction.php (hunk 2 above)
Comment (ciastektk) [reply src/bundle/Templating/Twig/Functions/AbstractFunction.php]: As we discussed, this approach will not change

**Pull request merged**